src/System.IO.Pipelines/Pipe.cs

```diff
@@ -322,8 +322,34 @@ internal void AdvanceWriter(int bytesWritten)
             } // and if zero, just do nothing; don't need to validate tail etc
         }
 
+        internal bool TryFlush(out FlushResult result)
+        {
+            // Always signal the reader
+            SignalReader();
```

Comment:
Also call it SignalReaderUnsyncronized do nobody uses it without lock by accident.